models/sudoswap/ethereum/sudoswap_ethereum_pools.sql

```diff
@@ -0,0 +1,275 @@
+{{
+ config(
+       alias='pools',
+       partition_by = ['block_date'],
+       materialized = 'incremental',
+       file_format = 'delta',
+       incremental_strategy = 'merge',
+       unique_key = ['pool_address', 'nft_contract_address', 'creator_address'],
+       post_hook='{{ expose_spells(\'["ethereum"]\',
+                                   "project",
+                                   "sudoswap",
+                                   \'["niftytable"]\') }}'
+      )
+}}
+
+WITH
+  pairs_created AS (
+    SELECT
+      CASE
+        WHEN _bondingCurve = '0x5b6ac51d9b1cede0068a1b26533cace807f883ee' THEN 'linear_bonding'
+        ELSE 'exp_bonding'
+      END as pricing_type,
+      _delta / 1e18 as delta,
+      CASE
+        WHEN _poolType = 0 THEN 'token'
+        WHEN _poolType = 1 THEN 'nft'
+        WHEN _poolType = 2 THEN 'trade'
+      END AS pool_type,
+      _spotPrice / 1e18 AS spot_price,
+      _nft AS nft_contract_address,
+      _initialNFTIDs AS nft_ids,
+      _fee AS initialfee,
+      output_pair AS pair_address,
+      call_block_time AS block_time,
+      tx.FROM AS creator_address,
+      date_trunc('day', now()) - date_trunc('day', call_block_time) AS days_passed,
+      cardinality(_initialNFTIDs) AS initial_nft_count,
+      tx.value / 1e18 AS initial_eth
+    FROM
+      {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }} cre
+      INNER JOIN {{ source('ethereum','transactions') }} tx ON tx.hash = cre.call_tx_hash
+    WHERE
+      call_success
+    {% if is_incremental() %}
+    -- this filter will only be applied on an incremental run. We only want to update with new transactions.
+    AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+  ),
+  most_recent_spot_delta AS (
+    SELECT
+      COALESCE(del.contract_address, spot.contract_address) AS pair_address,
+      del.delta,
+      spot.spot_price
+    FROM
+      (
+        SELECT
+          *
+        FROM
+          (
+            SELECT
+              contract_address,
+              newDelta / 1e18 AS delta,
+              row_number() over (
+                PARTITION BY
+                  contract_address
+                ORDER BY
+                  evt_block_number DESC,
+                  evt_index DESC
+              ) as most_recent
+            FROM
+              {{ source('sudo_amm_ethereum','LSSVMPair_general_evt_DeltaUpdate') }}
```

Comment:
i realize not all decoded tables need an incremental filter, just checking if it's needed here or not

Reply:
Im not sure, is it needed here?